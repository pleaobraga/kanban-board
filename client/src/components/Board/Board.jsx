import './Board.scss'

import React from 'react'
import PropTypes from 'prop-types'

import TaskList from '../TaskList'

export const Board = ({ TaskLists }) => {
  return (
    <div className="board">
      {TaskLists.map((list, index) => (
        <TaskList
          id={list.id}
          key={list.id}
          name={list.name}
          Cards={list.Cards}
          index={index}
        />
      ))}
    </div>
  )
}

Board.propTypes = {
  TaskLists: PropTypes.arrayOf(PropTypes.shape(TaskList.propTypes))
}

export default Board
