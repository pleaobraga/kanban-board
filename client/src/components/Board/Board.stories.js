import React from 'react'
import { storiesOf } from '@storybook/react'
import { Board } from './Board'

const cards = [
  {
    id: '0',
    index: 0,
    type: 'feature',
    duration: 1,
    severity: 'hight'
  },
  {
    id: '1',
    index: 1,
    type: 'research',
    duration: 2,
    severity: 'medium'
  },
  {
    id: '2',
    index: 2,
    type: 'update',
    duration: 3,
    severity: 'low'
  }
]

const lists = [
  {
    id: '0',
    name: 'Backlog',
    cards
  },
  {
    id: '1',
    name: 'To do ',
    cards
  },
  {
    id: '2',
    name: 'Done',
    cards
  }
]

export default storiesOf('Components | Board', module).add(
  'default',
  () => <Board lists={lists} />,
  { info: { inline: true, header: false } }
)
