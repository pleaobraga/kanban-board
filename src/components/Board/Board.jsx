import './Board.scss'

import React from 'react'
import PropTypes from 'prop-types'

import TaskList from '../TaskList'

export const Board = ({ lists }) => {
  return (
    <div className="board">
      {lists.map(list => (
        <TaskList
          id={list.id}
          key={list.id}
          title={list.title}
          cards={list.cards}
        />
      ))}
    </div>
  )
}

Board.propTypes = {
  lists: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      cards: PropTypes.arrayOf(
        PropTypes.shape({
          type: PropTypes.oneOf([
            'feature',
            'bugFix',
            'update',
            'research',
            'content'
          ]).isRequired,
          duration: PropTypes.number.isRequired,
          severity: PropTypes.oneOf(['hight', 'medium', 'low']).isRequired
        })
      )
    })
  )
}

export default Board
