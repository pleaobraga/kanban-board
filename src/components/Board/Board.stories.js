import React from 'react'
import { storiesOf } from '@storybook/react'
import { Board } from './Board'

const cards = [
  {
    type: 'feature',
    duration: 1,
    severity: 'hight'
  },
  {
    type: 'research',
    duration: 2,
    severity: 'medium'
  },
  {
    type: 'update',
    duration: 3,
    severity: 'low'
  }
]

const lists = [
  {
    title: 'Backlog',
    cards
  },
  {
    title: 'To do ',
    cards
  },
  {
    title: 'Done',
    cards
  }
]

export default storiesOf('Components | Board', module).add(
  'default',
  () => <Board lists={lists} />,
  { info: { inline: true, header: false } }
)
