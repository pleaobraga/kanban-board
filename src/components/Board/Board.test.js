import React from 'react'
import { shallow } from 'enzyme'
import Board from './Board'

const cards = [
  {
    id: '0',
    index: 0,
    type: 'feature',
    duration: 1,
    severity: 'hight'
  },
  {
    id: '1',
    index: 1,
    type: 'research',
    duration: 2,
    severity: 'medium'
  },
  {
    id: '2',
    index: 2,
    type: 'update',
    duration: 3,
    severity: 'low'
  }
]

const lists = [
  {
    id: '0',
    title: 'Backlog',
    cards
  },
  {
    id: '1',
    title: 'To do ',
    cards
  },
  {
    id: '2',
    title: 'Done',
    cards
  }
]

describe('Board', () => {
  const component = shallow(<Board lists={lists} />)

  it('render properly', () => {
    expect(component).toMatchSnapshot()
  })
})
