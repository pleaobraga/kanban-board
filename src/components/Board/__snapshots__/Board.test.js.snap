// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board render properly 1`] = `
<div
  className="board"
>
  <TaskList
    cards={
      Array [
        Object {
          "duration": 1,
          "id": "0",
          "index": 0,
          "severity": "hight",
          "type": "feature",
        },
        Object {
          "duration": 2,
          "id": "1",
          "index": 1,
          "severity": "medium",
          "type": "research",
        },
        Object {
          "duration": 3,
          "id": "2",
          "index": 2,
          "severity": "low",
          "type": "update",
        },
      ]
    }
    id="0"
    key="Backlog"
    title="Backlog"
  />
  <TaskList
    cards={
      Array [
        Object {
          "duration": 1,
          "id": "0",
          "index": 0,
          "severity": "hight",
          "type": "feature",
        },
        Object {
          "duration": 2,
          "id": "1",
          "index": 1,
          "severity": "medium",
          "type": "research",
        },
        Object {
          "duration": 3,
          "id": "2",
          "index": 2,
          "severity": "low",
          "type": "update",
        },
      ]
    }
    id="1"
    key="To do "
    title="To do "
  />
  <TaskList
    cards={
      Array [
        Object {
          "duration": 1,
          "id": "0",
          "index": 0,
          "severity": "hight",
          "type": "feature",
        },
        Object {
          "duration": 2,
          "id": "1",
          "index": 1,
          "severity": "medium",
          "type": "research",
        },
        Object {
          "duration": 3,
          "id": "2",
          "index": 2,
          "severity": "low",
          "type": "update",
        },
      ]
    }
    id="2"
    key="Done"
    title="Done"
  />
</div>
`;
